test/runner/filters/unified_filter.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+/**
+ * Filter for tests that require the unified topology
+ *
+ * example:
+ * metadata: {
+ *    requires: {
+ *      unifiedTopology: <boolean>
+ *    }
+ * }
+ */
+class UnifiedTopologyFilter {
+  filter(test) {
+    if (!test.metadata) return true;
+    if (!test.metadata.requires) return true;
+    if (!test.metadata.requires.unifiedTopology) return true;
```

Comment:
nit: might be neat to make this instead:

```js
if (typeof test.metadata.requires.unifiedTopology !== 'boolean') return true;

return test.metadata.requires.unifiedTopology === !!process.env.MONGODB_UNIFIED_TOPOLOGY;
```

so that we could do `{unifiedTopology: false}` for legacy tests

Reply:
good catch! I'm going to include this in the NODE-2274 work, because we need to skip tests there for unified topology (the opposite of what this filter lets us do)